### PR TITLE
ENG-1111 use go helm library

### PR DESCRIPTION
**Pull request by zreigz**
## Summary
Replace helm CLI with go client

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [ ] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

File: pkg/wkspace/minimal.go [hunk 1]
```diff
@@ -56,107 +69,131 @@ func FormatValues(w io.Writer, vals string, output *output.Output) (err error) {
...
 	}
-
-	out, err := output.Read(pathing.SanitizeFilepath(filepath.Join(root, app, "output.yaml")))
+	defaultVals, err := getValues(m.Name)
 	if err != nil {
-		out = output.New()
+		return err
+	}
+	namespace := m.Config.Namespace(m.Name)
+	actionConfig := new(action.Configuration)
```
Comment (michaeljguarino) [inline pkg/wkspace/minimal.go:82]: could we refactor this helm cli setup into a helper function?  I feel like it benefits from being reusable

File: pkg/wkspace/minimal.go [hunk 2]
```diff
@@ -56,107 +69,131 @@ func FormatValues(w io.Writer, vals string, output *output.Output) (err error) {
...
 
-	backup = fmt.Sprintf("%s.bak", valsFile)
-	err = os.Rename(valsFile, backup)
+	utils.Warn("helm upgrade --install --skip-crds --namespace %s %s %s %s\n", namespace, m.Name, path, strings.Join(extraArgs, " "))
+	chart, err := loader.Load(path)
 	if err != nil {
-		return
+		return err
 	}
+	client := action.NewUpgrade(actionConfig)
```
Comment (michaeljguarino) [inline pkg/wkspace/minimal.go:94]: shouldn't this be initialized after the below if?

File: pkg/wkspace/minimal.go [hunk 3]
```diff
@@ -56,107 +69,131 @@ func FormatValues(w io.Writer, vals string, output *output.Output) (err error) {
...
+	installManifest, err := getTemplate(m.Name, namespace, true, true)
+	if err != nil {
+		return err
 	}
+	currentSpecs := make(map[string]*diffmanifest.MappingResult)
+	var newSpecs map[string]*diffmanifest.MappingResult
+
+	currentSpecs = diffmanifest.Parse(string(releaseManifest), namespace, false, helm3TestHook, helm2TestSuccessHook)
+	newSpecs = diffmanifest.Parse(string(installManifest), namespace, false, helm3TestHook, helm2TestSuccessHook)
+	helmdiff.Manifests(currentSpecs, newSpecs, &helmdiff.Options{
```
Comment (michaeljguarino) [inline pkg/wkspace/minimal.go:189]: this is super nice to have statically linked now, nice

**Review by michaeljguarino — COMMENTED** [review]
We also need to cover the implementation in `plural destroy` which also calls some helm commands

**Comment by zreigz** [comment]
> We also need to cover the implementation in `plural destroy` which also calls some helm commands

@michaeljguarino  yes, I noticed the issue, I will introduce helm destroy to this PR

**Review by michaeljguarino — APPROVED** [review]
this looks a lot cleaner

**Pull request merged**